src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -2585,19 +2585,32 @@ public void PropertyFunctionGetFolderPath()
         }
 #endif
 
-#if FEATURE_RUNTIMEINFORMATION
         /// <summary>
-        /// Expand property function that uses types available only on .netstandard1.3 and above (RuntimeInformation)
-        /// The test exercises: static method invocation, static property invocation, method invocation expression as argument, call chain expression as argument,
-        /// </summary>
-        [Fact]
-        public void PropertyFunctionRuntimeInformation()
-        {
-            var pg = new PropertyDictionary<ProjectPropertyInstance>();
+        /// The test exercises: RuntimeInformation / OSPlatform usage, static method invocation, static property invocation, method invocation expression as argument, call chain expression as argument
+        /// </summary>
+        [Theory]
+        [InlineData(
+            "$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Create($([System.Runtime.InteropServices.OSPlatform]::$$platform$$.ToString())))))",
+            "True"
+        )]
+        [InlineData(
+            @"$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::$$platform$$)))",
```

Comment:
I think that doing that translation requires a bigger, non local code change than just a new intrinsic method `IsOSPlatform(string platform)` that contains all the logic. As @AndyGerlicher mentioned offline, we could blindly call `TryParse` on `OSPlatform` and if it does not parse, then throw an exception where we blindly enumerate the accepted enum values. This way we do not care about the underlying implementation. Does this sound good?